### PR TITLE
Adding MadCatz FLY5 Flight Stick

### PR DESCRIPTION
**Pull request by john-tiberius**
Here we go.

### CODE REVIEW

**Comment by dbankieris** [comment]
@john-tiberius Can I borrow one of these to double-check your decoding?

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,10 @@ project(IDF)
 
 option(ENABLE_CAN_DEVICES "Enable CAN controlled input devices" OFF)
 
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/CMakeListsDIR.txt)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/CMakeListsDIR.txt)
+endif()
+
```
Comment (dbankieris) [inline CMakeLists.txt]: I'm not well-versed in cmake. What's this bit about?

File: include/idf/MadCatz.hh [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+#include "idf/InputLayout.hh"
+#include "idf/SingleInput.hh"
+
+namespace idf {
+
+/**
+ * %MadCatz hand controller's input layout
+ *
+ * @author Derek Bankieris
```
Comment (dbankieris) [inline include/idf/MadCatz.hh]: You can put yourself here or just remove this line.

File: include/idf/UsbMadCatz.hh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "idf/UsbDevice.hh"
+#include "idf/MadCatz.hh"
+
+namespace idf {
+
+/**
+ * USB MadCatz hand controller
+ *
+ * @author Derek Bankieris
```
Comment (dbankieris) [inline include/idf/UsbMadCatz.hh]: You can put yourself here or just remove this line.

File: source/idf/UsbMadCatz.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+UsbMadCatz::UsbMadCatz(int vendorID, int productID, const std::string& id) :
+    UsbDevice(id, vendorID, productID, 8) {}
+
+UsbMadCatz::UsbMadCatz(int vendorID, int productID, const std::string& id, unsigned length) :
+    UsbDevice(id, vendorID, productID, length) {}
+
+void UsbMadCatz::decode(const std::vector<unsigned char>& data) {
+    uint16_t value;
+     
+    //forwardBackwardPivot.setValue(((unsigned)data[2] & 0xF) << 6 | data[1] >> 2);
```
Comment (dbankieris) [inline source/idf/UsbMadCatz.cpp]: Remove commented out code.

File: source/idf/UsbMadCatz.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    // 10-bits B23 B22 B21 B20 B17 B16 B15 B14 B13 B12 
+    value = ((uint16_t) data[2] & 0x000f) << 6 |  ((uint16_t)data[1] & 0x00FC) >> 2;
+    forwardBackwardPivot.setValue(value);
+
+    //leftRightPivot.setValue(((unsigned)data[1] & 3) << 8 | data[0]);
```
Comment (dbankieris) [inline source/idf/UsbMadCatz.cpp]: Remove commented out code.

**Comment by dbankieris** [comment]
The decoding looks great, nice work! The only thing I'm not sure about is `scrollUp` vs `scrollDown`. When I roll my thumb up the scroll wheel, `scrollDown` goes active. Are those accidentally swapped, or am I thinking about it wrong?

**Comment by john-tiberius** [comment]
They could be swapped, they should work as you suggest.

File: CMakeLists.txt (hunk 1 above)
Comment (dbankieris) [reply CMakeLists.txt]: @john-tiberius Hey, you! Answer my question!

**Pull request merged**